File: server/src/main/java/com/fluffy/exam/application/response/ExamDetailResponse.java
package com.fluffy.exam.application.response;

import java.time.LocalDateTime;
import java.util.List;

public record ExamDetailResponse(
        Long id,
        String title,
        String description,
        String status,
        List<QuestionResponse> questions,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {

    public interface QuestionResponse {
        Long id();

        String text();

        String passage();

        String type();
    }

    public record AnswerQuestionResponse(
            Long id,
            String text,
            String passage,
            String type
    ) implements QuestionResponse {
    }

    public record ChoiceQuestionResponse(
            Long id,
            String text,
            String passage,
            String type,
            List<QuestionOptionResponse> options
    ) implements QuestionResponse {

        public record QuestionOptionResponse(
                Long id,
                String text
        ) {
        }
    }
}

File: server/src/main/kotlin/com/fluffy/exam/application/response/ExamDetailResponse.kt
package com.fluffy.exam.application.response

import java.time.LocalDateTime

data class ExamDetailResponse(
    val id: Long,
    val title: String,
    val description: String,
    val status: String,
    val questions: List<QuestionResponse>,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
)

interface QuestionResponse {

    val id: Long
    val text: String
    val passage: String
    val type: String
}

data class AnswerQuestionResponse(
    override val id: Long,
    override val text: String,
    override val passage: String,
    override val type: String,
) : QuestionResponse

data class ChoiceQuestionResponse(
    override val id: Long,
    override val text: String,
    override val passage: String,
    override val type: String,
    val options: List<QuestionOptionResponse>,
) : QuestionResponse


data class QuestionOptionResponse(
    val id: Long,
    val text: String,
)
